tests/src/test/scala/whisk/core/containerpool/docker/test/DockerContainerTests.scala

```diff
@@ -0,0 +1,640 @@
+        val cpuShares = 1
+        val environment = Map("test" -> "hi")
+        val network = "testwork"
+        val name = "myContainer"
+
+        val container = DockerContainer.create(
+            transid = transid,
+            image = image,
+            memory = memory,
+            cpuShares = cpuShares,
+            environment = environment,
+            network = network,
+            name = Some(name))
+
+        await(container)
+
+        docker.pulls should have size 0
+        docker.runs should have size 1
+        docker.inspects should have size 1
+        docker.rms should have size 0
+
+        val (testImage, args) = docker.runs.head
+        testImage shouldBe "image"
+
+        // Assert fixed values are passed as well
+        args should contain allOf ("--cap-drop", "NET_RAW", "NET_ADMIN")
+        args should contain inOrder ("--ulimit", "nofile=1024:1024")
+        args should contain inOrder ("--pids-limit", "1024") // OW PR 2119
+
+        // Assert proper parameter translation
+        args should contain inOrder ("--memory", s"${memory.toMB}m")
+        args should contain inOrder ("--memory-swap", s"${memory.toMB}m")
+        args should contain inOrder ("--cpu-shares", cpuShares.toString)
+        args should contain inOrder ("--network", network)
+        args should contain inOrder ("--name", name)
+
+        // Assert proper environment passing
+        args should contain allOf ("-e", "test=hi", "SERVICE_IGNORE=true")
+    }
+
+    it should "pull a user provided image before creating the container" in {
+        implicit val docker = new TestDockerClient
+        implicit val runc = stub[RuncApi]
+
+        val container = DockerContainer.create(transid = transid, image = "image", userProvidedImage = true)
+        await(container)
+
+        docker.pulls should have size 1
+        docker.runs should have size 1
+        docker.inspects should have size 1
+        docker.rms should have size 0
+    }
+
+    it should "remove the container if inspect fails" in {
+        implicit val docker = new TestDockerClient {
+            override def inspectIPAddress(id: ContainerId, network: String)(implicit transid: TransactionId): Future[ContainerIp] = {
+                inspects += ((id, network))
+                Future.failed(new RuntimeException())
+            }
+        }
+        implicit val runc = stub[RuncApi]
+
+        val container = DockerContainer.create(transid = transid, image = "image")
+        a[WhiskContainerStartupError] should be thrownBy await(container)
+
+        docker.pulls should have size 0
+        docker.runs should have size 1
+        docker.inspects should have size 1
+        docker.rms should have size 1
+    }
+
+    it should "disambiguate errors if user images are provided" in {
+        implicit val docker = new TestDockerClient {
+            override def inspectIPAddress(id: ContainerId, network: String)(implicit transid: TransactionId): Future[ContainerIp] = {
```

Comment:
this test seems wrong in that unlike the next test that follows, failure on inspect is a system error not a black box error, is it not?